src/code/z_vimode.c

```diff
@@ -1,17 +1,214 @@
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_vimode/func_800AC030.s")
+void ViMode_LogPrint(OSViMode* viMode) {
+    OSViMode** viMode_ptr = &viMode;
+#define _LOG(line, fmt_str, val)                 \
```

Comment:
We already have the `LOG_ADDRESS` macro for this (defined [here](https://github.com/zeldaret/oot/blob/98aef8988adbcb5960e8001e70141d0d1a9034bd/include/macros.h#L72)).